src/notifiers/pool.js

```diff
@@ -0,0 +1,42 @@
+import _s from 'underscore.string';
+import glob from 'glob';
+import fs from 'fs';
+import path from 'path';
+
+import NotifierLoader from './loader';
+
+let _instance = null;
+
+export default class NotifierPool {
+  constructor() {
+    if (!_instance) {
+      this.notifiers = {};
+      this._loadNotifiers();
+
+      _instance = this;
+    }
+
+    return _instance;
+  }
+
+  getNotifier(className) {
```

Comment:
Since you can make several notifier with same className, i think it should be rather "id" if notifier
let's say if we make SlackNotifier, there could be
- WebTeam SlackNotifier
- TrackTicket SlackNotifier
- FrontEnd SlackNotifier